packages/grafana-ui/src/components/Table/TableNG/TableNG.test.tsx

```diff
@@ -0,0 +1,1135 @@
+      expect(filteredRowCount).toBe(1);
+
+      // Verify the visible row contains "A1"
+      const visibleCells = container.querySelectorAll('[role="gridcell"]');
+      const cellTexts = Array.from(visibleCells).map((cell) => cell.textContent);
+      expect(cellTexts).toContain('A1');
+      expect(cellTexts).not.toContain('A2');
+    });
+  });
+
+  describe('Resizing', () => {
+    it('calls onColumnResize when column is resized', () => {
+      const onColumnResize = jest.fn();
+
+      const { container } = render(
+        <TableNG
+          enableVirtualization={false}
+          data={createBasicDataFrame()}
+          width={800}
+          height={600}
+          onColumnResize={onColumnResize}
+        />
+      );
+
+      // Find resize handle
+      const resizeHandles = container.querySelectorAll('.rdg-header-row > [role="columnheader"] .rdg-resizer');
+
+      if (resizeHandles.length > 0) {
+        // Simulate resize by triggering mousedown, mousemove, mouseup
+        fireEvent.mouseDown(resizeHandles[0]);
+        fireEvent.mouseMove(resizeHandles[0], { clientX: 250 });
+        fireEvent.mouseUp(resizeHandles[0]);
+
+        // Check that onColumnResize was called
+        expect(onColumnResize).toHaveBeenCalled();
+      }
+    });
+  });
+
+  describe('Text wrapping', () => {
+    it('defaults to not wrapping text', () => {
+      const { container } = render(
+        <TableNG enableVirtualization={false} data={createBasicDataFrame()} width={800} height={600} />
+      );
+
+      const cells = container.querySelectorAll('[role="gridcell"]');
+      const cellStyles = window.getComputedStyle(cells[0]);
+      expect(cellStyles.getPropertyValue('white-space')).toBe('nowrap');
+    });
+
+    it('applies text wrapping styles when wrapText is true', () => {
+      // Create a frame with text wrapping enabled
+      const frame = createBasicDataFrame();
+      frame.fields.forEach((field) => {
+        if (field.config?.custom) {
+          field.config.custom.cellOptions = {
+            ...field.config.custom.cellOptions,
+            wrapText: true,
+          };
+        }
+      });
+
+      const { container } = render(
+        <TableNG
+          enableVirtualization={false}
+          data={frame}
+          width={800}
+          height={600}
+          fieldConfig={{
+            defaults: {
+              custom: {
+                cellOptions: {
+                  wrapText: true,
+                },
+              },
+            },
+            overrides: [],
+          }}
+        />
+      );
+
+      // Check for cells with wrap styling
+      const cells = container.querySelectorAll('[role="gridcell"]');
+      const cellStyles = window.getComputedStyle(cells[0]);
+
+      // In the getStyles function, when textWrap is true, whiteSpace is set to 'break-spaces'
+      expect(cellStyles.getPropertyValue('white-space')).toBe('break-spaces');
+    });
+  });
+
+  describe('Pagination navigation', () => {
```

Comment:
Minor request, but can we move this to or next to the existing pagination block?